tests/SpecTests/DocumentsMatchConstraintTest.php

```diff
@@ -146,13 +145,13 @@ public function errorMessageProvider()
                 ['foo' => ['foo' => 'bar', 'bar' => 'baz']],
             ],
             'Scalar value not equal' => [
-                'Field path "foo": Failed asserting that two values are equal.',
+                'Field path "foo": Failed asserting that two strings are equal.',
```

Comment:
Error messages change due to us no longer using a direct comparison for scalar values, and instead leaving those checks to PHPUnit comparators.